### PR TITLE
feat(crons): Add ability to mute monitor environments

### PR DESCRIPTION
**Pull request by rjo100**
Allows ability to `mute` MonitorEnvironments

Depends on https://github.com/getsentry/sentry/pull/62183

### CODE REVIEW

File: src/sentry/monitors/endpoints/organization_monitor_environment_details.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    )
+    def put(
+        self, request: Request, organization, project, monitor, monitor_environment
+    ) -> Response:
+        """
+        Update a monitor environment.
+        """
+
+        if request.data["status"] == MonitorStatus.DISABLED:
+            monitor_environment.update(status=MonitorStatus.DISABLED)
```
Comment (rjo100) [inline src/sentry/monitors/endpoints/organization_monitor_environment_details.py:59]: need to add setting to `ACTIVE` too oops

File: src/sentry/monitors/endpoints/organization_monitor_environment_details.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    owner = ApiOwner.CRONS
+
+    @extend_schema(
+        operation_id="Update a Monitor Environment",
+        parameters=[
+            GlobalParams.ORG_SLUG,
+            MonitorParams.MONITOR_SLUG,
+            MonitorParams.ENVIRONMENT,
+        ],
+        request=MonitorValidator,
```
Comment (rjo100) [inline src/sentry/monitors/endpoints/organization_monitor_environment_details.py:44]: this is probably wrong

File: src/sentry/monitors/endpoints/organization_monitor_environment_details.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        return self.respond(serialize(monitor, request.user))
+
+    @extend_schema(
+        operation_id="Delete a Monitor Environments",
+        parameters=[
+            GlobalParams.ORG_SLUG,
+            MonitorParams.MONITOR_SLUG,
+            MonitorParams.ENVIRONMENT,
+        ],
+        request=MonitorValidator,
```
Comment (rjo100) [inline src/sentry/monitors/endpoints/organization_monitor_environment_details.py:81]: this is probably wrong

**Comment by rjo100** [comment]
changing this to `is_muted` (forthcoming)

File: src/sentry/monitors/endpoints/organization_monitor_environment_details.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+            MonitorParams.ENVIRONMENT,
+        ],
+        responses={
+            202: RESPONSE_ACCEPTED,
+            401: RESPONSE_UNAUTHORIZED,
+            403: RESPONSE_FORBIDDEN,
+            404: RESPONSE_NOT_FOUND,
+        },
+    )
+    def delete(
```
Comment (davidenwang) [inline src/sentry/monitors/endpoints/organization_monitor_environment_details.py:86]: nice will switch over to this, once its all settled in

**Review by davidenwang — APPROVED** [review]
seems fine to me, does experimental publish to the api docs site, but just have kind of a warning given?

**Comment by rjo100** [comment]
not sure. will revisit later

**Pull request merged**